src/main/java/org/openbase/bco/bcozy/view/ObserverText.java

```diff
@@ -48,32 +60,40 @@
      * actual text
      */
     public ObserverText(final String identifier) {
-        super();
+        this();
         setIdentifier(identifier);
-        LanguageSelection.getInstance().addObserver(this);
+    }
+
+    @Override
+    public void update(final Observable observable, final Object arg) {
+        if (getIdentifier() == null) {
+            return;
+        }
+
+        String text = LanguageSelection.getLocalized(getIdentifier());
+
+        super.setText(applyOnNewText.apply(text));
     }
 
     /**
-     * Sets the new identifier for this ObserverText.
+     * Sets the new identifier for this ObserverLabel.
      *
      * @param identifier identifier
      */
-    public void setIdentifier(final String identifier) {
-        this.identifier = identifier;
-        languageBundle = ResourceBundle.getBundle(Constants.LANGUAGE_RESOURCE_BUNDLE, Locale.getDefault());
+    public void setIdentifier(String identifier) {
+        this.identifier.set(identifier);
+    }
 
-        try {
-            super.setText(languageBundle.getString(identifier));
-        } catch (MissingResourceException ex) {
-            if (!identifier.equals(Constants.DUMMY_LABEL)) {
-                ExceptionPrinter.printHistory("Could not resolve Identifier [" + identifier + "]!", ex, LOGGER, LogLevel.WARN);
-            }
-            super.setText(identifier);
-        }
+    public String getIdentifier() {
+        return identifier.get();
     }
 
-    @Override
-    public void update(final Observable observable, final Object arg) {
-        setIdentifier(identifier);
+    public SimpleStringProperty identifierProperty() {
+        return identifier;
+    }
+
+    public void setApplyOnNewText(Function<String, String> applyOnNewText) {
+        this.applyOnNewText = applyOnNewText != null ? applyOnNewText : Function.identity();
+        this.update(null, null);
```

Comment:
null should not be used as default argument.
If possible introduce a additional methods without arguments.